OpenRA.Mods.Common/SpriteLoaders/GenieSlpLoader.cs

```diff
@@ -54,14 +80,14 @@ static byte[] ReadRowCommands(Stream stream, GenieSlpFrameHeader header)
 					return rowData.ToArray();
 				}
 
-				var opcode = 0xf;
-				var rowPosX = 0;
+				int opcode;
+				var rowPosX = skipLeft;
 
 				do
 				{
 					var currByte = GetSingleByte(stream);
 
-					opcode = currByte & 0x0f;
+					opcode = currByte & 15;
```

Comment:
Generally, Hex is easier to understand when using logical and/or. Of course, bitfields would be best, but I don't think they're usable in C#.